src/Symfony/Component/VarDumper/Command/Descriptor/Descriptor.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Command\Descriptor;
+
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\VarDumper\Cloner\Data;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+abstract class Descriptor
+{
+    protected $io;
+
+    public function __construct(SymfonyStyle $io)
+    {
+        $this->io = $io;
+    }
+
+    abstract public function __invoke(Data $data, array $context, int $clientId): void;
```

Comment:
I'm not a big fan of `__invoke` in core code, it hides the underlying interface and remove the action to the reader

Reply:
reworked descriptors